src/dynamic_honey_badger/dynamic_honey_badger.rs

```diff
@@ -150,12 +148,15 @@ where
             .collect();
         let step = self
             .honey_badger
-            .propose(&InternalContrib {
-                contrib,
-                key_gen_messages,
-                votes: self.vote_counter.pending_votes().cloned().collect(),
-            }).map_err(Error::ProposeHoneyBadger)?;
-        self.process_output(step)
+            .propose(
+                rng,
+                &InternalContrib {
+                    contrib,
+                    key_gen_messages,
+                    votes: self.vote_counter.pending_votes().cloned().collect(),
+                },
```

Comment:
I'd construct the `InternalContrib` in a separate step now, to avoid too much nesting.